include/cuda/std/detail/libcxx/include/cstdlib

```diff
@@ -103,11 +103,7 @@ void *aligned_alloc(size_t alignment, size_t size);                       // C11
 #elif __has_builtin(__builtin_unreachable)
 #  define _LIBCUDACXX_UNREACHABLE() __builtin_unreachable()
 #else
-#ifdef __CUDA_ARCH__
-#  define _LIBCUDACXX_UNREACHABLE() __trap()
-#else
-#  define _LIBCUDACXX_UNREACHABLE() ::abort()
-#endif // __CUDA_ARCH__
+#  define _LIBCUDACXX_UNREACHABLE() NV_IF_ELSE_TARGET(NV_IS_HOST,__trap(),::abort())
```

Comment:
This might be critical, as it technically does not count as a single statement and so C++11 functions that should be constexpr cannot use it. That said there are non that I know of

Reply:
Isn't this logic backwards? I think it should be `NV_IF_ELSE_TARGET(NV_IS_HOST, ::abort(), __trap())`? 

On a different note, is "unreachable" the right name here? With a name like that, I'd expect it to use `__builtin_unreachable()`.

Reply:
Argh, you are absolutely right